src/memo/memo.ml

```diff
@@ -178,15 +240,47 @@ module Cached_value = struct
         t.calculated_at <- Run.current ();
         Some t.data
     end
+  and get_sync : type a. a t -> a option = fun t ->
+    if Run.is_current t.calculated_at then
+      Some t.data
+    else begin
+      let dep_changed = function
+        | Last_dep.T (node, prev_output) ->
+          match node.state with
+          | Running_sync run ->
+            if not (Run.is_current run)
+            then
+              true
+            else
+              failwith "dependency cycle: "
```

Comment:
How about:
```
if Run.is_current run then
  Exn.code_error "dependency_cycle" [];
true
```

Reply:
I prefer to keep the "else" if that's OK. (it makes it more obvious that the two branches are mutually exclusive)

Reply:
fine with me